execute/plugin_functions.go

```diff
@@ -226,11 +228,8 @@ func mergeMessageObservations(
 ) (exectypes.MessageObservations, error) {
 	// Create a validator for each chain
 	validators := make(map[cciptypes.ChainSelector]validation.MinObservationFilter[cciptypes.Message])
-	idFunc := func(data cciptypes.Message) [32]byte {
-		return sha3.Sum256([]byte(fmt.Sprintf("%v", data)))
-	}
```

Comment:
The "idFunc" is only used internally by the `MinObervationValidator`. This one is observing messages, so the messageID could be used.

There are 3 types being counted currently:
* cciptypes.Message
* exectypes.CommitData (MerkleRoot is not sufficient because this struct also includes messages and which messages are executed).
* NonceTriplet (simple container for source chainSel, sender and the nonce)